src/main/scala/nl.knaw.dans.easy.solr4files/components/AuthorisationComponent.scala

```diff
@@ -36,9 +36,8 @@ trait AuthorisationComponent extends DebugEnhancedLogging {
     private implicit val jsonFormats: Formats = DefaultFormats
 
     def getAuthInfoItem(bagId: UUID, path: Path): Try[AuthInfoItem] = {
+      val uri = baseUri.resolve(s"$bagId/${ escapePath(path) }")
```

Comment:
that is not what I meant to say :-)
`escapePath` can fail, right? If it does, now the exception flies off into where???

Reply:
The java code doesn't throw anything.

Reply:
copied from `com.google.common.escape`:

```scala
  /**
   * Returns the escaped form of a given literal string.
   *
   * <p>Note that this method may treat input characters differently depending on the specific
   * escaper implementation.
   *
   * <ul>
   * <li>{@link UnicodeEscaper} handles <a href="http://en.wikipedia.org/wiki/UTF-16">UTF-16</a>
   * correctly, including surrogate character pairs. If the input is badly formed the escaper should
   * throw {@link IllegalArgumentException}.
   * <li>{@link CharEscaper} handles Java characters independently and does not verify the input for
   * well formed characters. A {@code CharEscaper} should not be used in situations where input is
   * not guaranteed to be restricted to the Basic Multilingual Plane (BMP).
   * </ul>
   *
   * @param string the literal string to be escaped
   * @return the escaped form of {@code string}
   * @throws NullPointerException if {@code string} is null
   * @throws IllegalArgumentException if {@code string} contains badly formed UTF-16 or cannot be
   *     escaped for any other reason
   */
  public abstract String escape(String string);
```

this method is used in

```scala
def escapePath(path: Path): String = {
  path.asScala.map(_.toString).map(pathEscaper.escape).mkString("/")
}
```